prospero-common/src/main/java/org/wildfly/prospero/model/ProsperoConfig.java

```diff
@@ -52,9 +53,36 @@ public List<RepositoryRef> getRepositories() {
     }
 
     public void addChannel(ChannelRef channelRef) {
+        // Check that neither GAV nor URL of added channel is equal to GAVs or URLs of existing channels.
+        if (StringUtils.isNotBlank(channelRef.getGav())) {
```

Comment:
Couldn't this be replaced with `channelRef. getGavOrUrlString`? For both new channelRef and existing ones it should give either the gav or url that can be compared straight away?

Reply:
So the config containing channels with both GAV and URL are not completely forbidden yet.

I think the problematic case is when adding a URL-containing-channel, and an existing channel from metadata would have both GAV and URL (equal to the one that is to be added) defined. The URL would then be masked if we queried just `getGavOrUrlString()`.

I wanted to prevent repetition of any given GAV or URL in the config. Plus at this level it's not guaranteed that channel to be added also doesn't contain both GAV and URL. Maybe it's not needed to be as strict though.

Reply:
hmm that's an interesting scenario. We cannot guarantee that a URL based channelRef isn't the same as GAV.

Let's say we have `org.test:channel-one` and user adds `https://maven.repo/org/test/channel-one/1.0.0/channel-one-1.0.0-channel.yaml`. They're both the same channel (assuming there's no newer version), but there's no way for us to tell.

Reply:
Let's got with this for now and change the serialisation to allow either GAV or URL. I think that'll solve this issue :)